include/rocksdb/utilities/backupable_db.h

```diff
@@ -165,6 +165,33 @@ struct BackupableDBOptions {
   // db_session_id as a fallback.
   BackupTableNameOption share_files_with_checksum_naming;
 
+  // Option for custom checksum functions.
+  // When this option is nullptr, BackupEngine will use its default crc32c as
+  // the checksum function.
+  //
+  // When it is not nullptr, this option comes into effect only if DB has a
+  // custom checksum factory and this option is set to the same factory.
+  //
+  // Regardless of what is set for this option, BackupEngine will always use
+  // its default crc32c checksum for creating, verifying, or restoring backups.
+  //
+  // If this option is set to the DB checksum factory, BackupEngine will try
+  // to find the requested checksum function and use it to create, verify, or
+  // restore backups, in addition to the default crc32c checksum function.
+  // When the DB checksum factory is set to GetFileChecksumGenCrc32cFactory(),
+  // we recommend not setting this option to the same factory to avoid
```

Comment:
Sorry I didn't quite understand this restriction. What would it take to avoid extra work when a user configures both DB and backup custom checksum function factories with `GetFileChecksumGenCrc32cFactory()`?

Reply:
My considerations were as follows. (none of them are critical, just writing down my thoughts, and let me know yours :P)

* The default crc32c function of the backup engine is not exactly the same as `GetFileChecksumGenCrc32cFactory()`. The default backup crc32c is little-endian while `GetFileChecksumGenCrc32cFactory()` is big-endian. (But actually, this consideration is no longer relevant after I changed the backup crc32c value to be a hex-encoded string where I also convert it to a big-endian string. Should I change it back to a little-endian hex-encoded string?)

* We may use one of the two functions but it is not clear which one is better. Should we use the default backup crc32c function? Or should we assume that by setting the option to `GetFileChecksumGenCrc32cFactory()`, users want to overwrite the backup default crc32c function (with a slightly different implementation)?

* When this option is enabled, it is supposed to compute two checksums. So if for whatever reasons users want to compute another crc32c similar to the default by hacking this option, maybe we should allow it.

* The codes look cleaner and it may be easier to maintain. With some extra logic and checking in the codes, we can choose to use one of the two functions. But if for whatever reasons the implementation of `GetFileChecksumGenCrc32cFactory()` changes (such as changing endianness, adding bit-mask at the start and/or at the end), we will not need to change too much in the backup engine. (Maybe we should unify the crc32c functions for db and backup in the future?)

Reply:
> I would be OK with removing the special treatment (if we haven't already?)

The treatment is actually not removed but generalized: whenever the `file_checksum_gen_factory` in DB is not null, we will use the data to check hand-off checksums (when backup engine has the same factory and generator).

And yes, I think we should have the story that "setting this option is good" :) I'll remove this confusing inline comment.